### ISSUE TITLE
[proposal] gang strategy enhancement

### ISSUE THREAD

**Issue by LY-today** [koordinator-sh/koordinator#2244]
**What is your proposal**:
Add user wishes comments to the logic in the gang policy to prevent the gang policy from directly invalidating after the pod is destroyed and rebuilt.
`if gang.getGangMatchPolicy() == extension.GangMatchPolicyOnceSatisfied && gang.isGangOnceResourceSatisfied() {
		return nil
	}`

**Why is this needed**:
The gang policy immediately becomes invalid after the pod is destroyed and rebuilt. The newly created pod becomes running even when the total number of pods does not meet the MinRequiredNumber. However, some users think this does not meet their expectations. They believe that the semantics of the gang policy are still followed in this scenario. , so it is recommended to add user intention judgment logic

**Is there a suggested solution, if so, please add it**:
Annotation can be added to the podGroup, for example: gang.scheduling.koordinator.sh/close-history-evaluate. At this time, the rebuilt pod is destroyed. When the total number of pods does not meet the MinRequiredNumber, it is still pending, maintaining gang semantics.

ps：If the community thinks this capability suggestion is reasonable, I am willing to submit the corresponding PR.

**Comment by DaPengDiting** [comment]
I'm also experiencing this confusion, and I'm glad to see someone bringing it up. I would appreciate it if the author could take a look at this issue.

**Comment by Park-Jiyeonn** [comment]
I totally agree with this proposal. I've encountered similar issues where the gang policy behavior doesn't align with user expectations after a pod is destroyed and rebuilt. Keeping the gang semantics intact until the MinRequiredNumber is satisfied makes sense from a scheduling perspective. I support adding the user intention logic to address this.

Thank you for raising this issue, and I look forward to seeing it resolved!

**Comment by JBinin** [comment]
I have encountered the same problem as the author. This  proposal has been of great help to me. Thank you.

**Comment by saintube** [comment]
@LY-today Thanks for your suggestions. Feel free to self-assign if you are willing to contribute or have been working on it!

**Comment by LY-today** [comment]
> @LY-today Thanks for your suggestions. Feel free to self-assign if you are willing to contribute or have been working on it!

@saintube 
First of all, I would like to ask if you agree with this feature enhancement? Secondly, do you agree that adding annotation annotations can solve this problem? I hope we can communicate the design method in advance, so that the progress will be smoother.

ps：I am very willing to submit the corresponding PR, and I will develop it after the communication is completed.

**Comment by LY-today** [comment]
@saintube PR has been submitted, please check,#2246
